.gitlab-ci.yml

```diff
@@ -0,0 +1,215 @@
+# Copyright 2018 Red Hat, Inc.
+# Author: Jan Pokorny <jpokorny@redhat.com>
+# Part of pacemaker project
+# SPDX-License-Identifier: FSFAP
+
+variables:
+  GIT_DEPTH: "16"
+
+stages:
+  - buildenv
+  - build
+  - test
+
+.fedora-rawhide: &fedora-rawhide-anchor
+  tags:
+    - docker
+  image: fedora:rawhide
+
+.buildenv: &buildenv-anchor
+  stage: buildenv
+  variables:
+    GIT_STRATEGY: none
+
+.compilation: &compilation-anchor
+  only:
+   - pushes
+  # only since GitLab 11.4
+  #  changes:
+  #    - "configure.ac"
+  #    - "**/*.[ch]"
+  #    - "**/*.xml"
+  #    - "**/*.xsl"
+  #    - "**/Makefile.am"
+  #    - ".gitlab-ci.yml"
+
+buildenv:fedora-rawhide:info:
+  # primarily to gather some info about the image itself/container settings
+  <<: *buildenv-anchor  #extends: .buildenv
+  <<: *fedora-rawhide-anchor  #extends: .fedora-rawhide
+  script:
+   - cat /proc/meminfo | grep Free
+   - df -H
+   - lsblk
+   - nproc
+   - ulimit -a
+   - dnf --version
+
+buildenv:fedora-rawhide:rpms:
+  # primarily to download RPMs, but alas, "download" plugin not baked into image
+  <<: *buildenv-anchor  #extends: .buildenv
+  <<: *fedora-rawhide-anchor  #extends: .fedora-rawhide
+  cache:
+    key: "fedora-rawhide-dnf"
+    paths:
+      - ".cache/fedora-rawhide/dnf/rawhide-*"
+  script:
+    - mkdir -p .cache/fedora-rawhide/dnf
+    - cd .cache/fedora-rawhide
+    - mv dnf/rawhide-* /var/cache/dnf 2>/dev/null || true
```

Comment:
I'm not quite clear on how caching works. "mkdir" implies the directory might not exist at this point, in which case it will empty and this glob will fail.

Reply:
It took me quite some to figure out how best to leverage the slicing
unit of the overall routine, jobs, since each job is a tabula rasa.
The key is to move the important intermediate data using this cache
facility.

So instead of downloading RPM packages over and over, they got
downloaded once (sum of all requirements need to be specified
in the initial job), and then dnf installations in subsequent
jobs where the packages are actually needed won't need to touch
a single bit on the web, recache metsdata, making for a swift
preparation for the actual work within the job boundary.

Here, the directory will surely exist or:
- dnf has failed in the initial job, which contradicts with
  this job getting started
- something changed in the logic of dnf -- very unlikely
  and easy to reflect

Reply:
sounds good